# Homework 2 - Hypothesis Testing

Homework setup such as imports are mentioned below.

In [165]:
import pandas as pd

## Part 1

## Part 2

### Task 1: Import the Data
Use the pandas.read_csv() function to import the dataset. The data is contained in two files: realEstate1.csv and realEstate2.csv. After you import these files separately, concatenate them into one big dataframe. This pandas dataframe will be used for data exploration and linear regression.

In [166]:
# Import Data
re1 = pd.read_csv("realEstate1.csv")
re2 = pd.read_csv("realEstate2.csv")

# Concatenate Dataframe
data_list = [re1, re2]
df = pd.concat(data_list, ignore_index=True)
display(df)

ListNo                  Access  AdditInfo  Acres             ListAgt  \
0    1397571                Concrete        NaN   0.29          Liz Slager   
1    1394021                 Asphalt        NaN   0.01      Angela Carlson   
2    1393604                     NaN        NaN   0.11     Rachel Kitterer   
3    1393602                Concrete        NaN   0.14     Rachel Kitterer   
4    1393829                 Asphalt        NaN   0.01  T. Carter Maudsley   
..       ...                     ...        ...    ...                 ...   
343  1277128                Concrete        NaN   0.28          Liz Slager   
344  1265274                Concrete        NaN   0.01    PATRICK WISCOMBE   
345  1241781                Concrete        NaN   0.00     Jacqui Johnston   
346  1238399                     NaN        NaN   0.07        Grady Kohler   
347  1222911  Common Drive; Concrete        NaN   0.01       Jennifer Beck   

                              AirType  \
0               Central Air; Electric   
1               Central Air; Electric   
2               Central Air; Electric   
3               Central Air; Electric   
4               Central Air; Electric   
..                                ...   
343             Central Air; Electric   
344             Central Air; Electric   
345             Central Air; Electric   
346             Central Air; Electric   
347  Central Air; Electric; Heat Pump   

                                             Amenities  Area  BackDim  \
0    See Remarks; Cable Tv Available; Cable Tv Wire...   101      0.0   
1            Cable Tv Available; Electric Dryer Hookup   101      0.0   
2                Cable Tv Wired; Electric Dryer Hookup   101      0.0   
3                Cable Tv Wired; Electric Dryer Hookup   101      0.0   
4                                       Cable Tv Wired   101      0.0   
..                                                 ...   ...      ...   
343  Cable Tv Wired; Electric Dryer Hookup; Exercis...   101      0.0   
344  Electric Dryer Hookup; Exercise Room; Gas Drye...   101      0.0   
345                       Exercise Room; Swimming Pool   101      0.0   
346          Cable Tv Available; Electric Dryer Hookup   101      0.0   
347  Cable Tv Wired; Clubhouse; Electric Dryer Hook...   101      0.0   

                               Basemnt  ...  WinEvp1 WinEvp2 WinEvp3 WinEvp4  \
0    Daylight; Entrance; Full; Walkout  ...        0       0       0       0   
1                     None/Crawl Space  ...        0       0       0       0   
2                                 Full  ...        0       0       0       0   
3                                 Full  ...        0       0       0       0   
4                     None/Crawl Space  ...        0       0       0       0   
..                                 ...  ...      ...     ...     ...     ...   
343                   Partial; Walkout  ...        0       0       0       0   
344                   None/Crawl Space  ...        0       0       0       0   
345                   None/Crawl Space  ...        0       0       0       0   
346                   None/Crawl Space  ...        0       0       0       0   
347                   None/Crawl Space  ...        0       0       0       0   

                          Window    WithDrDt YearBlt    Zip Zoning     LastMod  
0                         Blinds  00/00/0000    1998  84103    NaN  08/29/2016  
1    Blinds; Plantation Shutters  00/00/0000    1965  84103    RES  08/22/2016  
2    Blinds; Plantation Shutters  00/00/0000    1940  84103    NaN  08/29/2016  
3                         Blinds  00/00/0000    1940  84103    NaN  08/29/2016  
4                            NaN  00/00/0000    1969  84103    NaN  08/18/2016  
..                           ...         ...     ...    ...    ...         ...  
343                 Blinds; Full  00/00/0000    1978  84103    RES  10/06/2015  
344                 Full; Shades  00/00/0000    1979  84103    NaN  05/24/2016  
345      

### Task 2: Clean the Data
There are 206 different variables associated with each of the 348 houses in this dataset. Skim them and try to get a rough understanding of what information this dataset contains. If you've never seen a real estate listing before, you might take a look at one on this website (Links to an external site.) to get a better sense of the meanings of the column headers in the dataset.

Only keep houses with List Price between 200,000 and 1,000,000 dollars. This is an arbitrary choice and we realize that some of you are high rollers, but for our purposes we'll consider the others as outliers.

Remove columns that you don't think contribute to the value of the house. This is a personal decision - what attributes of a house are important to you? You should at least keep the following variables since the questions below will use them: ['Acres', 'Deck', 'GaragCap', 'Latitude', 'Longitude', 'LstPrice', 'Patio', 'PkgSpacs', 'PropType', 'SoldPrice', 'Taxes', 'TotBed', 'TotBth', 'TotSqf', 'YearBlt']

Check the datatypes and convert any numbers that were read as strings to numerical values. (Hint: You can use str.replace() (Links to an external site.) to work with strings.) If there are any categorical values you're interested in, then you should convert them to numerical values as we saw in the notes (the get_dummies function may help). In particular,

Convert 'TotSqf' to an integer.
Add a new column titled Prop_Type_SingleFamily that is set to 0 if the i-th listing is a condo or townhouse, or 1 if the i-th listing is a single family house.
Remove the listings with erroneous 'Longitude' (one has Longitude = 0) and 'Taxes' values (two have unreasonably large values).

In [167]:
df = df[(df.LstPrice >= 200000) & (df.LstPrice <= 1000000)] # Filter Data
df.reset_index(inplace=True, drop=True) # Fix index

# Filter columns
df = df[['Acres', 'Deck', 'GaragCap', 'Latitude', 'Longitude', 'LstPrice', 'Patio', 'PkgSpacs', 'PropType', 'SoldPrice', 'Taxes', 'TotBed', 'TotBth', 'TotSqf', 'YearBlt']]

# Fix dtypes
df.TotSqf = pd.Series(df.TotSqf).str.replace(',', '', regex=True)
df.TotSqf = pd.to_numeric(df.TotSqf)

Prop_Type_SingleFamily_data = []

for i in df['PropType']:
    if i == 'Single Family':
        Prop_Type_SingleFamily_data.append(1)
    else:
        Prop_Type_SingleFamily_data.append(0)

df['Prop_Type_SingleFamily'] = Prop_Type_SingleFamily_data

df = df[(df.Longitude != 0) & (df.Taxes <= 10000) & (df.Taxes >= 1000)] # Filter Data
df.reset_index(inplace=True, drop=True) # Fix index
df

Acres  Deck  GaragCap   Latitude   Longitude  LstPrice  Patio  PkgSpacs  \
0     0.29     1         3  40.779524 -111.893542    725000      1         0   
1     0.01     0         0  40.773243 -111.883291    205000      0         0   
2     0.11     0         0  40.773972 -111.867470    390000      1         0   
3     0.14     0         1  40.783009 -111.872100    445000      1         3   
4     0.01     0         2  40.771693 -111.886951    339000      0         2   
..     ...   ...       ...        ...         ...       ...    ...       ...   
249   0.28     1         3  40.782933 -111.851143    995000      0         6   
250   0.01     0         2  40.770904 -111.886201    349999      2         2   
251   0.00     0         0  40.773843 -111.891911    249900      0         0   
252   0.07     0         1  40.772835 -111.854867    330000      0         0   
253   0.01     1         2  40.771980 -111.885262    385000      1         0   

          PropType  SoldPrice  Taxes  TotBed  TotBth  TotSqf  YearBlt  \
0    Single Family     752000   7244     4.0     5.0    5415     1998   
1            Condo     205000   1431     2.0     1.0     924     1965   
2    Single Family     390000   1816     3.0     1.0    1702     1940   
3    Single Family     443000   3260     3.0     2.0    1756     1940   
4            Condo     334000   2141     3.0     2.0    1368     1976   
..             ...        ...    ...     ...     ...     ...      ...   
249  Single Family     949000   6658     6.0     4.0    5740     1978   
250          Condo     340000   1877     2.0     2.0    1473     1979   
251          Condo     231000   2812     2.0     2.0    1385     1976   
252  Single Family     310000   2079     5.0     3.0    2094     1886   
253          Condo     370000   3644     2.0     2.0    2028     1981   

     Prop_Type_SingleFamily  
0                         1  
1                         0  
2                         1  
3                         1  
4                         0  
..                      ...  
249                       1  
250                       0  
251                       0  
252                       1  
253                       0  

[254 rows x 16 columns]

### Task 3: Exploratory Data Analysis
Explore the dataset. Write a short description of the dataset describing the number of items, the number of variables and check to see if the values are reasonable.

In [168]:
df.describe()

Acres        Deck    GaragCap    Latitude   Longitude  \
count  254.000000  254.000000  254.000000  254.000000  254.000000   
mean     0.132323    0.681102    1.283465   40.777405 -111.875359   
std      0.124623    0.798143    0.973138    0.005784    0.013801   
min      0.000000    0.000000    0.000000   40.769424 -111.913645   
25%      0.050000    0.000000    0.250000   40.772358 -111.886521   
50%      0.110000    1.000000    1.000000   40.775630 -111.874036   
75%      0.160000    1.000000    2.000000   40.781845 -111.865921   
max      0.690000    5.000000    5.000000   40.791059 -111.843083   

             LstPrice       Patio    PkgSpacs      SoldPrice        Taxes  \
count      254.000000  254.000000  254.000000     254.000000   254.000000   
mean    468331.744094    0.633858    1.035433  455283.622047  3083.917323   
std     200026.738539    0.685677    1.588846  189861.027774  1664.854395   
min     200000.000000    0.000000    0.000000  190000.000000  1100.000000   
25%     315000.000000    0.000000    0.000000  310000.000000  1897.000000   
50%     419000.000000    1.000000    0.000000  401500.000000  2472.000000   
75%     598000.000000    1.000000    2.000000  570000.000000  3861.750000   
max    1000000.000000    3.000000    8.000000  980000.000000  9205.000000   

           TotBed      TotBth       TotSqf      YearBlt  \
count  254.000000  254.000000   254.000000   254.000000   
mean     3.228346    2.543307  2503.708661  1940.468504   
std      1.242775    1.076195  1223.088956    35.887354   
min      1.000000    0.000000   844.000000  1852.000000   
25%      2.000000    2.000000  1550.500000  1908.000000   
50%      3.000000    2.000000  2272.500000  1937.000000   
75%      4.000000    3.000000  3081.000000  1976.000000   
max      8.000000    7.000000  7508.000000  2015.000000   

       Prop_Type_SingleFamily  
count              254.000000  
mean                 0.748031  
std                  0.435000  
min                  0.000000  
25%                  0.250000  
50%                  1.000000  
75%                  1.000000  
max                  1.000000

The dataset has exactly 254 entries and has 14 variables. I found that there was an entry with tax = 1, so I filtered the data above in Task 2, to have the lowest tax value as 1000. Apart from that the data seems to be pretty reasonable.